server.js

```diff
@@ -95,7 +95,7 @@ app.prepare().then(() => {
         )
       })
       res.end()
-    } else if (/^\/doc.*/i.test(pathname)) {
+    } else if (/^\/doc(?!s\/).*/i.test(pathname)) {
       // path /doc*/... -> /doc/...
```

Comment:
this is not true anymore, right? 

what is the current semantics of this redirect?

Reply:
It is. Except for /docs/* paths. Updated note in 361849e.

Another option is to revert this change and move the .md files from public/docs/ to somewhere else that doesn't overlap with this redirect, maybe public/md/ ?

Reply:
I mean in 43303c0

Reply:
so, we have an extremely strange combination now, let's cleanup it somehow.

docs redirects to doc
docs/ does not redirect
documentation redirects
docsddfsersfasdsaer redirects

let's just kill this rule for simplicity? I think it's outdated (when we had the previous domain - dataversioncontrol.com) anyway and can be handler by the cloudflare.

Also it's a bit strange to have `/docs/something.md` and /doc/something` I liked the previous location - it was clear that it's static - `static/docs/...`

Reply:
/doctor-pepper redirects 😋 

Ok, I killed the redirect now but had to leave the rule for the part that enforces 404 response codes. See 5f9deb25.

> I liked the previous location - it was clear that it's static

Yeah but it's now [deprecated](https://github.com/zeit/next.js/blob/master/errors/static-dir-deprecated.md) from Node. public/ is pretty obvious too, I think.

Reply:
location itself is fine, it's confusing to have `dvc.org/docs/get-started/index.md` and `dvc.org/doc/get-started` ... and the only differentiator is `s` between valid path prefixes ... it was way better to have exactly `dvc.org/static/docs/get-started/index.md`

Reply:
I mean. I can undo this whole change but I'm just following best practices here. Also, your example `/docs/get-started/index.md` is a static asset, not a routed path in our docs engine, like `/doc/get-started`. We never link to static assets directly anyway (with or without static/), so it won't be confusing to users, just maybe to us as developers (or possibly to other contributors). That's why I suggested moving the public/docs/ folder to something like public/md/ , but up to you 🙂

Reply:
I think it should be `public/static/*`:

https://nextjs.org/blog/next-9-1

```
Because public also covers the static directory use case we have decided to deprecate the static directory in favor of creating a public/static folder with the same functionality.
```

it means that public is not a replacement for static.

Reply:
OK, will do. (In fact that was an intermediate step in this PR but I didn't see need for the additional directory back then.)

And restore the /doc*/{subpath} -> /doc/{subpath} redirect? Or leave that out?

Reply:
It makes some sense to me to have `docs -> doc`, `documentation -> doc` and normalization `doc/` -> `doc` (actually we probably need this for any valid doc/* URL).

Reply:
OK I reinstated the redirect and moved public/docs/ to public/static/docs/, fixing all references accordingly (including some previously broken ones).

> normalization doc/ -> doc (actually we probably need this for any valid doc/* URL)

Good idea. I also implemented this via 301 redirects. Is that OK? I remember we've had problems with 301 redirects before... (They're very hard to undo.)

Note that trailing slashes in regular Next pages like /features/ result in a 404. Should we care about this inconsistency? Or perhaps implement this in our docs engine instead of 301 redirects to remove the trailing slash?

Reply:
p.s. re 30x redirects: refer to the conversation in https://github.com/iterative/dvc.org/issues/757#issuecomment-549494538.

Reply:
I guess the trailing slash behavior inconsistency is not a big deal. Resolving